include/envoy/http/header_map.h

```diff
@@ -111,6 +111,12 @@ class HeaderString {
   void setInteger(uint64_t value);
 
   /**
+   * Set the value of the string to a static string reference.
+   * @param static_value MUST point to static data.
```

Comment:
Yes. Let me update the comments to make that more clear. Would you suggest a different function name than that which includes "static" ? (Not sure what might be better).